File: bigdata/src/main/java/com/hl/bigdata/flink/stream/java/MyPartition.java
package com.hl.bigdata.flink.stream.java;

import org.apache.flink.api.common.functions.Partitioner;

/**
 * 自定义分区
 * @author huanglin
 * @date 2024/07/03 17:48
 */
public class MyPartition implements Partitioner<Integer> {

    @Override
    public int partition(Integer key, int i) {
        System.out.println("分区总数: " + i);

        if(key % 2 == 0) {
            return 0;
        } else {
            return 1;
        }
    }
}

File: bigdata/src/main/java/com/hl/bigdata/flink/stream/scala/MyPartition.scala
package com.hl.bigdata.flink.stream.scala

import org.apache.flink.api.common.functions.Partitioner

/**
 * @author huanglin
 * @date 2024/07/04 11:34
 */
class MyPartition extends Partitioner[Int]{

  override def partition(k: Int, i: Int): Int = {
    println("分区总数: " + i)
    if(k % 2 == 0) {
      0
    } else {
      1
    }
  }
}
